examples/developer-tools/src/blocks/input.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as Blockly from 'blockly/core';
+
+/**
+ * Check to see if more than one input has this name.
+ * If so, applies a warning to the block.
+ * 
+ * Highly inefficient (On^2), but n is small.
+ * @param referenceBlock Block to check.
+ */
+const inputNameCheck = function (referenceBlock: Blockly.Block) {
+  if (referenceBlock.isDeadOrDying()) {
+    // Block has been deleted.
+    return;
+  }
+  const name = referenceBlock.getFieldValue('INPUTNAME').toLowerCase();
+  let count = 0;
+  const blocks = referenceBlock.workspace.getAllBlocks(false);
+  for (const block of blocks) {
+    const otherName = block.getFieldValue('INPUTNAME');
+    if (
+      block.isEnabled() &&
+      !block.getInheritedDisabled() &&
+      otherName &&
+      otherName.toLowerCase() === name
+    ) {
+      count++;
+    }
+  }
+  const msg =
+    count > 1 ? 'There are ' + count + ' input blocks\n with this name.' : null;
+  referenceBlock.setWarningText(msg);
+};
+
+// Inputs that should take a "type" input for connection checks
+const inputsWithTypeInputs = new Set(['input_value', 'input_statement']);
```

Comment:
Ehh. I think the majority of the reason this is confusing it because of the use of the word "input", not the word "type". Adding inputs to an input block is confusing. The word type isn't overloaded here. If I changed it here, I'd also want to change the name of the "type" blocks, but I think it would be even more confusing to have the user-facing text say "type" (which matches all of our documentation) but have the block definition code use the language "connection check." So I don't think this would really reduce the confusion, and there just is bound to be a little bit when we're talking about making blocks out of blocks.

Reply:
Does "type" match all of our documentation? I've been working to change it over to use "connection check" or "check", since connection checks != types.

Reply:
Sorry, I meant "type" matches the documentation & video for the block factory. I do think there is room for confusion by changing it since the other blocks more or less correlate to the old blocks pretty well. But I did change it anyways. I don't think it makes this code any less confusing tbh, but it will be more consistent with our other block definition documentation.